### PR TITLE
update KFP v2 migration guide to include typecasting breaking change

### PR DESCRIPTION
**Pull request by connor-mccarthy**

### CODE REVIEW

File: content/en/docs/components/pipelines/v2/migration.md [hunk 1]
```diff
@@ -444,6 +444,36 @@ def pipeline():
...
+    learning_rate: float,
+):
+    print(f"number_of_epochs={number_of_epochs}")
+    print(f"learning_rate={learning_rate}")
+
+
+def training_pipeline(number_of_epochs: int = 1):
+    train(
+        number_of_epochs=number_of_epochs,
+        learning_rate="0.1",
```
Comment (chensun) [inline content/en/docs/components/pipelines/v2/migration.md:471]: nit: add an inline comment here to say this would be incompatible in v2, and what would be compatible in v2, so it's a little bit easier to the readers?

**Comment by chensun** [comment]
/lgtm

Letting you control when to add approval label.

**Review by chensun — APPROVED** [review]
/lgtm
/approve
